[darrenchooji] iP

darrenchooji opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

darrenchooji committed:
Level-0

darrenchooji committed:
Level-0

darrenchooji committed:
Level-1

darrenchooji committed:
Level-0 done

darrenchooji committed:
Level-1 done

darrenchooji committed:
Level-1 done

darrenchooji committed:
Level-2 done

darrenchooji committed:
Level-3 Done

darrenchooji committed:
Level-3 Done

darrenchooji committed:
Level-4 Done

darrenchooji committed:
A-TextUiTesting done

darrenchooji committed:
A-TextUiTesting done

darrenchooji committed:
Level-5 done

darrenchooji committed:
Level-6 Done

darrenchooji committed:
A-Enums done

darrenchooji committed:
A-Enums done

darrenchooji committed:
Level-7 done

darrenchooji committed:
Merge branch 'branch-Level-7'

darrenchooji committed:
Level-8 Implemented Date

darrenchooji committed:
Level-8 Done

darrenchooji committed:
Merge branch 'branch-Level-8'

darrenchooji committed:
A-MoreOOP done

darrenchooji committed:
Merge branch 'branch-A-MoreOOP'

darrenchooji committed:
A-Packages done

darrenchooji committed:
Merge branch 'branch-A-Packages'

darrenchooji committed:
Merge remote-tracking branch 'origin/add-gradle-support'

darrenchooji committed:
A-Gradle done

darrenchooji committed:
A-JUnit done

darrenchooji committed:
A-Jar

darrenchooji committed:
JavaDocs Fiona.java

darrenchooji committed:
JavaDocs fiona.task

darrenchooji committed:
A-JavaDocs done

darrenchooji committed:
Merge branch 'branch-A-JavaDoc'

darrenchooji committed:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

darrenchooji committed:
A-CodingStandard done

darrenchooji committed:
Merge branch 'branch-A-CodingStandard'

darrenchooji committed:
A-CodingStandard updated

darrenchooji committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

darrenchooji committed:
Level-9 completed

darrenchooji committed:
Level-9 completed

darrenchooji committed:
Merge branch 'branch-Level-9'

darrenchooji committed:
Level-9 completed

darrenchooji committed:
A-Checkstyle

darrenchooji committed:
Level-10 almost completed. Left with the bye/termination GUI feature.

darrenchooji committed:
Level-10 done

darrenchooji committed:
Merge branch 'branch-Level-10'

darrenchooji committed:
A-Varargs done

themintchoco commented on src/main/java/fiona/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return {@code true} if the task is completed, {@code false} otherwise.
+     */
+    public boolean getIsDone() {
+        return this.isDone;
+    }
```
Methods that return booleans should be named to sound like booleans

themintchoco reviewed (commented):
LGTM! Just one nit about method naming. Great work!

OvOtter commented on data/fiona.txt:
```diff
@@ -0,0 +1,2 @@
+T | 0 | CS2103T iP
```
Save files should not be uploaded to github (put in gitignore!)

OvOtter commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,31 @@
+    @Override
+    public void start(Stage stage) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
```
I like the good OOP practice here :)

OvOtter commented on src/main/java/fiona/command/Action.java:
```diff
@@ -0,0 +1,56 @@
+     * @param action The string representation of an action.
+     * @return The corresponding {@code Action} enum value, or {@code UNKNOWN} if invalid.
+     */
+    public static Action fromString(String action) {
```
Maybe the name of this method can be "getActionFromString"?

OvOtter reviewed (commented):
lgtm!

darrenchooji committed:
Update build.gradle

Ensure that the jar file, 'fiona.jar' can be build using gradle.

darrenchooji committed:
Update Task.java

Include `assert` to verify assumptions or invariants in the code.

darrenchooji committed:
Update Fiona.java

Shortened handleCommand method and findTask method to improve readability

darrenchooji committed:
Update Fiona.java

darrenchooji committed:
Update Fiona.java

Shortened handleCommand and findTasks method to improve readability of the code.

darrenchooji committed:
Update Fiona.java

Use streams in listTasks and findTasksByKeyword.

darrenchooji committed:
Merge branch 'master' into branch-A-CodeQuality

darrenchooji committed:
Merge branch 'master' into branch-A-Streams

darrenchooji committed:
Update Fiona.java

Shortened methods to improve readability.

darrenchooji committed:
Merge branch 'branch-A-CodeQuality'

darrenchooji committed:
Merge branch 'branch-A-Streams'

darrenchooji committed:
Updated Fiona.java

darrenchooji committed:
Updated Fiona.java and Parser.java

Updated the search function so that user can search by date-time and date.

darrenchooji committed:
Merge branch 'branch-C-BetterSearch'

darrenchooji committed:
Updated *.css and *.fxml files.

Create a new look for the GUI to make it more visually appealing.

darrenchooji committed:
Updated Deadline.java, Event.java, TaskList.java, and Fiona.java.

Ensure users do not enter past Deadlines or past Events.

darrenchooji committed:
Updated Event.java

Updated Event to ensure users cannot enter past events.

darrenchooji committed:
More testings

darrenchooji committed:
Updated README.md

Provided a user guide.

darrenchooji committed:
Updated files

darrenchooji committed:
Updated files

darrenchooji committed:
Update DeadlineTest.java

Altered testing deadline dates to a later date in 2035 to ensure that the same test commands can be reused in the future until 2035.

darrenchooji committed:
Update EventTest.java

Altered testing deadline dates to a later date in 2035 to ensure that the same test commands can be reused in the future until 2035.

darrenchooji committed:
Update FionaTest.java

Altered testing dates to a later date in 2035 to ensure that the same test commands can be reused in the future until 2035.

darrenchooji committed:
Update Fiona.java

darrenchooji committed:
Update Ui.java
